node-builder: add virtio-blk kata pod sandboxing and separate out tardev-snapshotter build

miz060 opened the pull request:
###### Merge Checklist  <!-- REQUIRED -->
- [ ] Followed patch format from upstream recommendation: https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format
  - [ ] Included a single commit in a given PR - at least unless there are related commits and each makes sense as a change on its own.
- [ ] Aware about the PR to be merged using "create a merge commit" rather than "squash and merge" (or similar)
- [ ] The `upstream/missing` label (or `upstream/not-needed`) has been set on the PR. 

###### Summary <!-- REQUIRED -->
- Add runtime config for virtio-blk kata pod sandboxing.
- Adjust dependencies for virtio-blk kata pod sandboxing.
- Extract tardev-snapshotter build and install into separate scripts.

###### Test Methodology
- image: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=766526&view=results
- kata conformance test: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=766880&view=results
- manual test passed for virtio-blk based kata pod sandboxing.

ms-mahuber commented on tools/osbuilder/rootfs-builder/cbl-mariner/config.sh:
```diff
@@ -6,6 +6,5 @@ OS_NAME=cbl-mariner
 OS_VERSION=${OS_VERSION:-3.0}
 LIBC="gnu"
 PACKAGES="kata-packages-uvm"
-[ "$CONF_GUEST" = yes ] && PACKAGES+=" kata-packages-uvm-coco"
```
I don't think we should remove this. Let's keep it - only for now, the set of packages would be common.

miz060 replied:
Got it. Added back

ms-mahuber commented on src/runtime/config/configuration-clh-snp.toml.in:
```diff
@@ -137,7 +137,7 @@ default_maxmemory = @DEFMAXMEMSZ@
 #   - virtio-fs-nydus
 #   - none
-shared_fs = "@DEFSHAREDFS_CLH_SNP_VIRTIOFS@"
+shared_fs = "none"
```
we should still leave the variables, to adapt the variables in our package_build.sh accordingly

miz060 replied:
Separated virtio-blk kata pod-sandboxing into a separate config.

ms-mahuber commented on tools/osbuilder/node-builder/azure-linux/package_build.sh:
```diff
@@ -41,29 +41,27 @@ fi
-if [ "${CONF_PODS}" == "yes" ]; then
-	agent_make_flags+=" AGENT_POLICY=yes"
-fi
+#if [ "${CONF_PODS}" == "yes" ]; then
```
Let's talk - we should factor our tardev into its own targets etc.

miz060 replied:
Separated tardev related builds into a separate script.

ms-mahuber commented on tools/osbuilder/node-builder/azure-linux/package_install.sh:
```diff
@@ -29,16 +28,15 @@ mkdir -p "${PREFIX}/${SHIM_CONFIG_PATH}"
-	cp -a --backup=numbered src/tardev-snapshotter/target/release/tardev-snapshotter ${PREFIX}/usr/bin/
-	mkdir -p ${PREFIX}/usr/lib/systemd/system/
-	cp -a --backup=numbered src/tardev-snapshotter/tardev-snapshotter.service ${PREFIX}/usr/lib/systemd/system/
+echo "Installing utarfs and kata-overlay binaries"
```
Why is this not factored into the tardev snapshotter scripting and sub-package?

miz060 replied:
Good point! Moved to new tardev install script.

ms-mahuber commented on tools/osbuilder/node-builder/azure-linux/package_build.sh:
```diff
@@ -41,29 +41,22 @@ fi
-	pushd src/overlay/
-	make all
-	popd
+echo "Building utarfs binary"
```
should also be moved to tardev part?

miz060 replied:
Moved to a separate tardev build script.

ms-mahuber commented on tools/osbuilder/node-builder/azure-linux/package_install.sh:
```diff
@@ -14,7 +14,6 @@ CONF_PODS=${CONF_PODS:-no}
 PREFIX=${PREFIX:-}
```
We should generally deploy a secondary virtio-blk config file which can be used with containerd + tardev-snapshotter

miz060 replied:
Good point. Added new config for virtio-blk based kata pod-sandboxing that uses tardev-snapshotter.

miz060 committed:
node-builder: add virtio-blk kata pod sandboxing and separate out tardev-snapshotter build

This PR is to support new virtio-blk kata pod sandboxing runtime. It
adds new runtime config, adjusts dependencies for kata pod sandboxing in
general, and extract tardev-snapshotter build and install steps into
separate scripts.

Signed-off by: Mitch Zhu <mitchzhu@microsoft.com>

manuelh-dev reviewed (changes requested):
See comments, vote to split. On hold for now.

ms-mahuber commented:
Note: To add required packages for kata UVM to list of packages (config.sh) as long as these are not found on the PMC meta-package
